src/useq/_mda_sequence.py

```diff
@@ -375,20 +390,27 @@ def iter_sequence(sequence: MDASequence) -> Iterator[MDAEvent]:
             x_pos = getattr(position, "x", None)
             y_pos = getattr(position, "y", None)
 
-        try:
-            z_pos = (
-                sequence._combine_z(_ev[Z][1], index[Z], channel, position)
-                if Z in _ev
-                else position.z
-                if position
-                else None
-            )
-        except sequence._SkipFrame:
+        if position and position.sequence:
+            for sub_event in iter_sequence(position.sequence):
+                if (
+                    position.sequence.grid_plan
+                    and position.sequence.grid_plan.is_relative
+                ):
+                    sub_event = sub_event.shifted(x_pos=x_pos, y_pos=y_pos)
+
+                if position.sequence.z_plan and position.sequence.z_plan.is_relative:
```

Comment:
what about subpositions with absolute plans?  imagine each position specified a different Z-top-bottom.  i think that would have worked before but not now?

add a test with two subpositions having `z_plan: {top: ..., bottom: ...}`